runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/refresh/ActiveWorkRefresher.java

```diff
@@ -64,14 +72,23 @@ public ActiveWorkRefresher(
       Supplier<Collection<ComputationState>> computations,
       DataflowExecutionStateSampler sampler,
       ScheduledExecutorService activeWorkRefreshExecutor,
-      Consumer<Map<HeartbeatSender, Heartbeats>> heartbeatSender) {
+      HeartbeatTracker heartbeatTracker) {
     this.clock = clock;
     this.activeWorkRefreshPeriodMillis = activeWorkRefreshPeriodMillis;
     this.stuckCommitDurationMillis = stuckCommitDurationMillis;
     this.computations = computations;
     this.sampler = sampler;
     this.activeWorkRefreshExecutor = activeWorkRefreshExecutor;
-    this.heartbeatSender = heartbeatSender;
+    this.heartbeatTracker = heartbeatTracker;
+    this.fanOutActiveWorkRefreshExecutor =
+        Executors.newCachedThreadPool(
+            new ThreadFactoryBuilder()
+                // Work refresh runs as a background process, don't let failures crash
+                // the worker.
+                .setUncaughtExceptionHandler(
```

Comment:
Let's remove this, sendHeartbeats is not throwing exceptions now

Reply:
done